src/agent/src/rpc.rs

```diff
@@ -238,13 +240,23 @@ impl AgentService {
 
         ctr.exec()?;
 
+        if sid == cid {
+            return Ok(());
+        }
+
         // start oom event loop
-        if sid != cid && ctr.cgroup_manager.is_some() {
-            let cg_path = ctr.cgroup_manager.as_ref().unwrap().get_cg_path("memory");
-            if cg_path.is_some() {
-                let rx = notifier::notify_oom(cid.as_str(), cg_path.unwrap()).await?;
+        if let Some(ref ctr) = ctr.cgroup_manager {
+            let cg_path = ctr.get_cg_path("memory");
+
+            if let Some(cg_path) = cg_path {
+                let rx = notifier::notify_oom(cid.as_str(), cg_path.to_string()).await?;
+
                 s.run_oom_event_monitor(rx, cid.clone()).await;
+            } else {
+                return Err(anyhow!("cannot get cgroup path"));
             }
+        } else {
+            return Err(anyhow!("cannot get cgroup manager"));
```

Comment:
This is changing behaviour, not just removing the `unwrap`, though I can't immediately tell which behaviour is correct: AFAICT the old version would just carry on in the `!ctr.cgroup_manager.is_some()` and `!cg_path.is_some()` cases, rather than returning an error.